### PR TITLE
Block Header Synchronization

### PR DESCRIPTION
**Pull request by ToWorld**
Support the ability, as follows:
1.Anchor Block
2.Forking handle
3.Maintain the latest block state
4.VerifyTx

## Description

What is the purpose of the change?
For the use of crossChain between xuperchain and xuperchain.

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Regression Test

### CODE REVIEW

File: core/contractsdk/cpp/example/xuper_relayer.cc [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+        return;
+    }
+}
+
+DEFINE_METHOD(XuperRelayer, verifyTx) {
+    xchain::Context* ctx = self.context();
+    const std::string blockid = ctx->arg("blockid");
+    const std::string txid = ctx->arg("txid");
+    const std::string proofPathStr = ctx->arg("proofPath");
+    const int txIndex = atoi(ctx->arg("txIndex").c_str());
```
Comment (yucaowang) [inline core/contractsdk/cpp/example/xuper_relayer.cc]: We do not verify the raw transaction data?

File: core/contractsdk/cpp/example/xuper_relayer.cc [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+bool within3Confirms(xchain::Context* ctx, const std::string& blockid, const std::string tipBlockid) {
+    int i = 0;
+    std::string currBlockid = tipBlockid;
+    while (i < 3) {
+        if (currBlockid == blockid) {
+            return false;
+        }
+        std::string blockHeaderStr;
+        const std::string currBlockHeaderKey = std::string(blockHeaderBucket) + currBlockid;
+        relayer::InternalBlock* blockHeader = new(relayer::InternalBlock);
```
Comment (fxsjy) [inline core/contractsdk/cpp/example/xuper_relayer.cc]: should this pointer be free later ?

File: core/contractsdk/cpp/example/xuper_relayer.cc (hunk 1 above)
Comment (ToWorld) [reply core/contractsdk/cpp/example/xuper_relayer.cc]: Issue about verify will be added later.

File: core/contractsdk/cpp/example/xuper_relayer.cc (hunk 2 above)
Comment (ToWorld) [reply core/contractsdk/cpp/example/xuper_relayer.cc]: Yes, I will corrected it.

File: core/contractsdk/cpp/example/xuper_relayer.cc [hunk 3]
```diff
@@ -0,0 +1,478 @@
...
+        }
+        std::string blockHeaderStr;
+        const std::string currBlockHeaderKey = std::string(blockHeaderBucket) + currBlockid;
+        std::unique_ptr<relayer::InternalBlock> blockHeader(new relayer::InternalBlock);
+        if (!ctx->get_object(currBlockHeaderKey, &blockHeaderStr)) {
+            return false;
+        }
+        blockHeader->ParseFromString(blockHeaderStr);
+        const std::string preHashStr = blockHeader->pre_hash();
+        std::string preHash = std::string(64, 'o');
```
Comment (0311xuyang) [inline core/contractsdk/cpp/example/xuper_relayer.cc]: hash value, not (64, '0') ?

File: core/contractsdk/cpp/example/xuper_relayer.cc [hunk 4]
```diff
@@ -0,0 +1,478 @@
...
+    }
+    std::string visualPreHash = std::string(64, 'o');
+    if (!encode(preHashBuf, visualPreHash)) {
+        ctx->error("encode preHash failed");
+        return;
+    }
+    // 判断preBlockcHeader是否存在
+    const std::string preBlockHeaderKey = std::string(blockHeaderBucket) + visualPreHash;
+    std::string preBlockHeaderStr;
+    std::unique_ptr<relayer::InternalBlock> preBlockHeader(new relayer::InternalBlock);
```
Comment (0311xuyang) [inline core/contractsdk/cpp/example/xuper_relayer.cc]: need to parse preBlockHeader and update the preBlockHeader value?

File: core/contractsdk/cpp/example/xuper_relayer.cc (hunk 3 above)
Comment (ToWorld) [reply core/contractsdk/cpp/example/xuper_relayer.cc]: Whatever, it's a kind of placeholder.

File: core/contractsdk/cpp/example/xuper_relayer.cc (hunk 4 above)
Comment (ToWorld) [reply core/contractsdk/cpp/example/xuper_relayer.cc]: Yes, it's necessary. We should verify it by contract self.

**Pull request merged**